server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java

```diff
@@ -349,89 +352,25 @@ public DfsLogger(ServerResources conf, String filename, String meta) throws IOEx
-        }
+        String cryptoStrategyClass = input.readUTF();
+        EncryptionStrategy encryptionStrategy = EncryptionStrategyFactory.setupReadEncryption(conf,
+            cryptoStrategyClass, EncryptionStrategy.Scope.WAL);
+        log.debug("Using {} for decrypting WAL", encryptionStrategy.getClass().getSimpleName());
+        decryptingInput = encryptionStrategy instanceof NoEncryptionStrategy ? input
+            : new DataInputStream(encryptionStrategy.decryptStream(input));
       } else {
+        log.error("Unsupported WAL version.");
         input.seek(0);
-        byte[] magicV2 = DfsLogger.LOG_FILE_HEADER_V2.getBytes(UTF_8);
-        byte[] magicBufferV2 = new byte[magicV2.length];
-        input.readFully(magicBufferV2);
-
-        if (Arrays.equals(magicBufferV2, magicV2)) {
-          // Log files from 1.5 dump their options in raw to the logger files. Since we don't know
-          // the class
-          // that needs to read those files, we can make a couple of basic assumptions. Either it's
-          // going to be
-          // the NullCryptoModule (no crypto) or the DefaultCryptoModule.
-
-          // If it's null, we won't have any parameters whatsoever. First, let's attempt to read
-          // parameters
-          Map<String,String> opts = new HashMap<>();
-          int count = input.readInt();
-          for (int i = 0; i < count; i++) {
-            String key = input.readUTF();
-            String value = input.readUTF();
-            opts.put(key, value);
-          }
-
-          if (opts.size() == 0) {
-            // NullCryptoModule, we're done
-            decryptingInput = input;
-          } else {
-
-            // The DefaultCryptoModule will want to read the parameters from the underlying file, so
-            // we will put the file back to that spot.
-            // @formatter:off
-            org.apache.accumulo.core.security.crypto.CryptoModule cryptoModule =
-              org.apache.accumulo.core.security.crypto.CryptoModuleFactory
-                .getCryptoModule(DefaultCryptoModule.class.getName());
-            // @formatter:on
-
-            CryptoModuleParameters params = CryptoModuleFactory
-                .createParamsObjectFromAccumuloConfiguration(conf);
-
-            // go back to the beginning, but skip over magicV2 already checked earlier
-            input.seek(magicV2.length);
-            params.setEncryptedInputStream(input);
-
-            params = cryptoModule.getDecryptingInputStream(params);
-            if (params.getPlaintextInputStream() instanceof DataInputStream) {
-              decryptingInput = (DataInputStream) params.getPlaintextInputStream();
-            } else {
-              decryptingInput = new DataInputStream(params.getPlaintextInputStream());
-            }
-          }
-
-        } else {
-
-          input.seek(0);
-          decryptingInput = input;
-        }
-
+        decryptingInput = input;
```

Comment:
I kept this because LogSorter uses both input and decryptingInput... although I am not sure why